cfme/services/myservice/ssui.py

```diff
@@ -242,14 +251,17 @@ def launch_vm_console(self, catalog_item):
 def retire(self):
     view = navigate_to(self, 'Retire')
     view.retire.click()
-    view = self.create_view(MyServicesView)
-    assert wait_for(
-        lambda: view.is_displayed, delay=3, num_sec=300,
-        message="waiting for view to be displayed"
-    )
+    if self.appliance.version < "5.10":
+        view = self.create_view(MyServicesView)
+    else:
+        view = self.create_view(DetailsMyServiceView)
+    view.wait_displayed()
     # TODO - remove sleep when BZ 1518954 is fixed
     time.sleep(10)
```

Comment:
Optional:
@niyazRedhat can you check if there is anything to remove this?

Reply:
Removed time.sleep

Reply:
awesome :+1: